src/AzureAuth/CommandMain.cs

```diff
@@ -255,11 +255,21 @@ private int GetToken()
         {
             try
             {
-                ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
+                TokenResult tokenResult = null;
+                ITokenFetcher tokenFetcher = null;
+
+                // When running multiple AzureAuth processes with the same tenant and resource ID,
+                // They may prompt many times, which is annoying and unexpected.
+                // Use GlobalLock to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = "Global\\" + this.Tenant + this.Resource;
+                using (new GlobalLock(lockName))

```

Comment:
After discussing with Tucker, I think I agree with the overall sentiment that `Mutex` provides what we need directly, and we don't have to test it's behavior. We can change this to use Mutex directly, and call it's Wait method - also using the `WaitOne(TimeSpan)` method to also enable a Wait Timeout here as a guard against unlimited blocking.

```suggestion
                using (var authLock = new Mutex(lockName))

```

Reply:
Just before we wait for the lock - we also want to add a `this.logger.LogDebug` call to signal a wait is about to happen to help debug scenarios where the auth cli might look like it is hanging. We could include the lock name in that debug message as well.